lib/performance_data/statistics.rb

```diff
@@ -0,0 +1,247 @@
+require 'json'
+require 'gds_api/performance_platform/data_out'
+
+module PerformanceData
+  class Statistics
+    ALLOWED_DOC_TYPES = ['smart_answer', 'simple_smart_answer'].freeze
+
+    attr_accessor :data_out, :slug, :part_urls, :is_multipart,
+                  :searches, :page_views, :problem_reports, :search_terms
+
+    def initialize(content, slug)
+      @data_out = GdsApi::PerformancePlatform::DataOut.new("http://www.performance.service.gov.uk")
+      @slug = slug
+      @part_urls = get_part_urls(content, slug)
+      @is_multipart = @part_urls.any? || ALLOWED_DOC_TYPES.include?(content["document_type"])
+    end
+
+    def searches
+      response = data_out.searches(slug, @is_multipart)
+      @searches ||= parse_values(response, "searchUniques:sum")
```

Comment:
To make this memoization do the right thing we'll need to do the following:

```ruby
def searches
  @searches ||= begin
    response = data_out.searches(slug, @is_multipart)
    parse_values(response, "searchUniques:sum")
  end
end
```

or just make the whole thing oneline:

```ruby
def searches
  @searches ||= parse_values(data_out.searches(slug, @is_multipart), "searchUniques:sum")
end
```

If we don't do this then we always make the call to `data_out.searches` and only avoid `parse_values` when we really want to avoid both once we've done them once.

We'll need to do the same for `page_views` and `problem_reports` too.  We do need to do it for `search_terms` as well, but it's a more complex method so we'll have to do something slightly more complex to make sure we avoid the work once we've called the method once.  Often I do something like:

```ruby
# before
def do_a_thing
  # lots of slow complicated code
  @do_a_thing = whatever_the_result_is
end

# after
def do_a_thing
  @do_a_thing ||= calculate_result_of_doing_a_thing
end

private
def calculate_result_of_doing_a_thing
  # lots of slow complicated code
  return whatever_the_result_is
end
```